lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +156,19 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_available_replicas
+      max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
+      if max_unavailable =~ /%/
+        (@desired_replicas * (100 - max_unavailable.to_i) / 100.0).ceil
+      else
```

Comment:
So if `max_unavailable` is nil, we'll effectively return `@desired_replicas`. If that's on purpose, I think it might be worth a comment. Because of the possibility of defaulting (see my other comment), we can't really validate this at template parsing time. However, it could be worth having a separate case that prints a warning (just once, not on every polling cycle). 

I think the main time this would happen is for deployments with `strategy: Recreate`, which I hadn't been considering previously. The only time I'm aware of us using them is single-replica deployments with PVCs, but there's nothing preventing others from using them with larger deployments. That would be a compelling use case for having the `%/#` option on the annotation as you suggested. That said, I still think `maxUnavailable` is the correct option for `rollingUpdate` deployments, and `%/#` options could be added in future PRs instead of now.

Reply:
I've added a validation that `'kubernetes-deploy.shopify.io/required-rollout = maxUnavailable` isn't in use with `strategy: Recreate`. Since that's a bit nonSensical.

Reply:
Works for me. The strategy isn't necessarily present in the templates, but the default has been RollingUpdate for many versions, so a validation using the definition should catch all cases in practice.